### PR TITLE
Test rmt feature includes enable/disable, sync, mirror and import

### PR DESCRIPTION
**Pull request by tinawang123**
To test rmt feature includes enable/disable, sync, mirror and
import SMT data to RMT

- Related ticket:
   https://progress.opensuse.org/issues/69166
   https://progress.opensuse.org/issues/69265
   https://progress.opensuse.org/issues/69271
   https://progress.opensuse.org/issues/69277
- Verification run: https://openqa.suse.de/tests/4546827#

### CODE REVIEW

File: tests/x11/rmt/rmt_feature.pm [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+sub run {
+    x11_start_program('xterm -geometry 150x35+5+5', target_match => 'xterm');
+    # Avoid blank screen since smt sync needs time
+    turn_off_gnome_screensaver;
+    become_root;
+    rmt_wizard();
+    # sync from SCC
+    rmt_sync;
+    # enable all modules of products
+    assert_script_run("rmt-cli p e SLES/15.2/x86_64 --all-modules");
```
Comment (lemon-suse) [inline tests/x11/rmt/rmt_feature.pm:32]: hard-code 15.2 is not make sense, if you don't want to import from setting, at least we can set it as variable.

File: tests/x11/rmt/rmt_feature.pm [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    assert_script_run("rmt-cli product list | grep sle-module-basesystem/15.2/x86_64");
+    assert_script_run("rmt-cli products enable --all-modules SLES/15");
+    assert_script_run("rmt-cli product list | grep sle-module-server-applications/15/aarch64");
+    assert_script_run("rmt-cli product list | grep sle-module-server-applications/15/ppc64le");
+    assert_script_run("rmt-cli product list | grep sle-module-server-applications/15/s390x");
+    assert_script_run("rmt-cli product list | grep sle-module-server-applications/15/x86_64");
+    assert_script_run("rmt-cli products disable 1946 1963 1967 1971 1982 1988 1992 1955 2075 1939 1950 1998 1973 1974 1976");
```
Comment (lemon-suse) [inline tests/x11/rmt/rmt_feature.pm:39]: It's hard for guest to understand the number, will the number changed later or will it keep stable?

File: tests/x11/rmt/rmt_feature.pm [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    turn_off_gnome_screensaver;
+    become_root;
+    rmt_wizard();
+    # sync from SCC
+    rmt_sync;
+    # enable all modules of products
+    assert_script_run("rmt-cli p e SLES/15.2/x86_64 --all-modules");
+    assert_script_run("rmt-cli product list | grep sle-module-basesystem/15.2/x86_64");
+    assert_script_run("rmt-cli products enable --all-modules SLES/15");
+    assert_script_run("rmt-cli product list | grep sle-module-server-applications/15/aarch64");
```
Comment (lemon-suse) [inline tests/x11/rmt/rmt_feature.pm:35]: Same for 15.

File: tests/x11/rmt/rmt_feature.pm [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    assert_script_run("rmt-cli repo disable 3393 3391");
+
+    # mirror packages
+    rmt_enable_pro;
+    rmt_mirror_repo();
+    assert_script_run("rmt-cli product list | grep sle-module-legacy/15/x86_64");
+
+    # import smt data
+    my $datapath = "/rmtdata/";
+    my $datafile = get_var("SMT_DATA_FILE");
```
Comment (lemon-suse) [inline tests/x11/rmt/rmt_feature.pm:66]: Where is the SMT data come from? Already upload to OSD?

File: tests/x11/rmt/rmt_feature.pm [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    my $dataurl  = get_var("SMT_DATA_URL");
+    assert_script_run("mkdir -p $datapath");
+    assert_script_run("cd $datapath");
+    assert_script_run("wget -q " . $dataurl . $datafile);
+    assert_script_run("tar -xzvf $datafile -C $datapath");
+    assert_script_run("chown -R _rmt:nginx $datapath");
+    assert_script_run("sudo rmt-data-import -d /rmtdata/smt-data-export/");
+
+    # check imported data
+    assert_script_run("rmt-cli repo enable 4205 4203");
```
Comment (coolgw) [inline tests/x11/rmt/rmt_feature.pm:76]: same question for this fix number

File: tests/x11/rmt/rmt_feature.pm (hunk 1 above)
Comment (tinawang123) [reply tests/x11/rmt/rmt_feature.pm:32]: This part only test --all-modules. Like others 'rmt-cli product enable 1798 1973 1974' just test rmt product enable feature.
This part only test product can enable 15.2 x86_64 all modules

File: tests/x11/rmt/rmt_feature.pm (hunk 4 above)
Comment (tinawang123) [reply tests/x11/rmt/rmt_feature.pm:66]: I put the SMT data to one server. We can get SMT data through 'wget'. Not OSD

File: tests/x11/rmt/rmt_feature.pm (hunk 2 above)
Comment (tinawang123) [reply tests/x11/rmt/rmt_feature.pm:39]: Same as above. We just test feature. The number will not change

**Comment by tinawang123** [comment]
Verified: https://openqa.suse.de/tests/4546827#

File: tests/x11/rmt/rmt_feature.pm [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    my $pro1 = get_var("PRODUCT_ALLMA");
+    assert_script_run("rmt-cli products enable --all-modules $pro1");
+    assert_script_run("rmt-cli product list | grep " . $pro1 . "/aarch64");
+    assert_script_run("rmt-cli product list | grep " . $pro1 . "/ppc64le");
+    assert_script_run("rmt-cli product list | grep " . $pro1 . "/s390x");
+    assert_script_run("rmt-cli product list | grep " . $pro1 . "/x86_64");
+    my @proid = split(/\n/, script_output("rmt-cli products list | awk -F '|' '{print \$2}'"));
+    diag "my output: @proid";
+    foreach my $i (@proid) {
+        script_run("rmt-cli products disable $i") if ($i =~ /\d{4}/);
```
Comment (lemon-suse) [inline tests/x11/rmt/rmt_feature.pm:45]: Why not assert_script_run?

File: tests/x11/rmt/rmt_feature.pm [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    # enable product with product ID 1798-Web and Scripting Module/15.1/x86_64
+    # 1973-Web and Scripting Module/15.2/aarch64 1974-Web and Scripting Module/15.2/ppc64le
+    assert_script_run("rmt-cli product enable 1798 1973 1974");
```
Comment (lemon-suse) [inline tests/x11/rmt/rmt_feature.pm:50]: Is there any pre-condition to do this? Need mirror all products?

Comment (tinawang123) [reply tests/x11/rmt/rmt_feature.pm:50]: Random ID

File: tests/x11/rmt/rmt_feature.pm [hunk 8]
```diff
@@ -0,0 +1,97 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
```
Comment (lemon-suse) [inline tests/x11/rmt/rmt_feature.pm:3]: 2019-2020

File: tests/x11/rmt/rmt_feature.pm [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    become_root;
+    rmt_wizard();
+    # sync from SCC
+    rmt_sync;
+    # enable all modules of products at one arch
+    my $pro = get_var("PRODUCT_ALLM");
+    assert_script_run("rmt-cli p e $pro --all-modules");
+    assert_script_run("rmt-cli product list | grep $pro");
+    # enable all modules of products at four arch
+    my $pro1 = get_var("PRODUCT_ALLMA");
```
Comment (coolgw) [inline tests/x11/rmt/rmt_feature.pm:36]: i suggest we combine two new setting to one, such as RMT_TEST_PRODUCT=15.2

Comment (tinawang123) [reply tests/x11/rmt/rmt_feature.pm:36]: As discussed in the meeting. Not change now.

**Pull request merged**